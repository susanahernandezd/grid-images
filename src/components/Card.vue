<template>
  <b-col v-if="card" cols="4" md="3" xl="2" class="mb-4">
    <div class="card" @click="removeCard">
      <div v-if="removing" class="card__spinner">
        <b-spinner label="Loading"></b-spinner>
      </div>
      <b-img class="card__image img-responsive" fluid :src="card.url" :alt="card.title"></b-img>
      <div class="card__text">
        <h2 class="card__title d-none d-md-block">{{ card.title }}</h2>
        <button class="card__button">
          <span class="sr-only">Remove photo</span>
        </button>
      </div>
    </div>
    <h2 class="card__title--sm d-md-none">{{ card.title }}</h2>
  </b-col>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  name: 'Card',
  data: () => ({
    removing: false
  }),
  props: {
    card: {
      type: Object,
      required: true
    }
  },
  methods: {
    ...mapActions(['deleteCard']),
    async removeCard () {
      if (!this.removing) {
        this.removing = true
        try {
          await this.deleteCard(this.card.id)
        } finally {
          this.removing = false
        }
      }
    }
  }
}
</script>

<style lang="scss">
@import './src/styles/components/_card';
</style>
