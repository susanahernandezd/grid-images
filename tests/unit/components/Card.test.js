import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import photosResponse from '#/fixtures/services/content/photosResponse'
import { BCol, BImg, BSpinner } from 'bootstrap-vue'
import Card from '@/components/Card'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Card Should', () => {
  let wrapper

  beforeEach(() => {
    const store = new Vuex.Store({
      state: {
        loading: {
          delete: 0
        }
      }
    })
    wrapper = shallowMount(Card, {
      stubs: {
        'b-col': BCol,
        'b-img': BImg,
        'b-spinner': BSpinner
      },
      localVue,
      store,
      propsData: {
        card: photosResponse[0]
      }
    })
  })

  it('display card image and title', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
