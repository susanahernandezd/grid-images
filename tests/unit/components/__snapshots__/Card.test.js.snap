// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should display card image and title 1`] = `
<div class="mb-4 col-md-3 col-xl-2 col-4">
  <div class="card">
    <!----> <img src="https://via.placeholder.com/600/92c952" alt="accusamus beatae ad facilis cum similique qui sunt" class="card__image img-responsive img-fluid">
    <div class="card__text">
      <h2 class="card__title d-none d-md-block">accusamus beatae ad facilis cum similique qui sunt</h2> <button class="card__button"><span class="sr-only">Remove photo</span></button>
    </div>
  </div>
  <h2 class="card__title--sm d-md-none">accusamus beatae ad facilis cum similique qui sunt</h2>
</div>
`;
